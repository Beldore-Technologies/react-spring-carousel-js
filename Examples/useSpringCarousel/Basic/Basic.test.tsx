import { render } from '@testing-library/react'
import { mockedItems } from '../mocked-data'
import { Basic } from './Basic'

describe('Basic example', () => {
  test('Snapshot', () => {
    const { container } = render(<Basic />)

    expect(container).toMatchSnapshot()
  })

  test('Render use-spring-carousel-wrapper', () => {
    const { getByTestId } = render(<Basic />)
    const carouselWrapper = getByTestId('use-spring-carousel-wrapper')

    expect(carouselWrapper).toBeInTheDocument()
  })

  test('Render use-spring-carousel-animated-wrapper', () => {
    const { getByTestId } = render(<Basic />)
    const carouselAnimatedWrapper = getByTestId(
      'use-spring-carousel-animated-wrapper',
    )

    expect(carouselAnimatedWrapper).toBeInTheDocument()
  })

  test('Render carousel items', () => {
    const { getAllByTestId } = render(<Basic />)
    const items = getAllByTestId('use-spring-carousel-item-wrapper')

    expect(items).toHaveLength(mockedItems.length)
  })

  test('Carousel wrapper has correct default styles', () => {
    const { getByTestId } = render(<Basic />)
    const carouselWrapper = getByTestId('use-spring-carousel-wrapper')

    expect(carouselWrapper).toHaveStyle({
      display: 'flex',
      position: 'relative',
      width: '100%',
      height: '100%',
      overflow: 'hidden',
    })
  })

  test('Carousel animated wrapper has correct default styles', () => {
    const { getByTestId } = render(<Basic />)
    const carouselAnimatedWrapper = getByTestId(
      'use-spring-carousel-animated-wrapper',
    )

    expect(carouselAnimatedWrapper).toHaveStyle({
      display: 'flex',
      flexDirection: 'row',
      top: 0,
      left: 0,
      width: '100%',
      height: '100%',
      position: 'relative',
    })
  })

  test('Carousel item has correct default styles', () => {
    const { getAllByTestId } = render(<Basic />)
    const items = getAllByTestId('use-spring-carousel-item-wrapper')

    items.forEach(i => {
      expect(i).toHaveStyle({
        display: 'flex',
        flex: `1 0 calc(100% / 1)`,
        position: 'relative',
      })
    })
  })
})
