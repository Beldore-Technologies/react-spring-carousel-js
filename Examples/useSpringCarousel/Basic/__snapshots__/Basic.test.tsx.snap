// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic example Snapshot 1`] = `
<div>
  <div
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
  >
    <div
      data-testid="use-spring-carousel-wrapper"
      style="display: flex; position: relative; width: 100%; height: 100%; overflow: hidden;"
    >
      <div
        data-testid="use-spring-carousel-animated-wrapper"
        style="display: flex; flex-direction: row; top: 0px; left: 0px; width: 100%; height: 100%; position: relative; transform: none;"
      >
        <div
          data-testid="use-spring-carousel-item-wrapper"
          style="display: flex; flex: 1 0 calc(100% / 1); position: relative;"
        >
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            Item 1
          </div>
        </div>
        <div
          data-testid="use-spring-carousel-item-wrapper"
          style="display: flex; flex: 1 0 calc(100% / 1); position: relative;"
        >
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            Item 2
          </div>
        </div>
        <div
          data-testid="use-spring-carousel-item-wrapper"
          style="display: flex; flex: 1 0 calc(100% / 1); position: relative;"
        >
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            Item 3
          </div>
        </div>
        <div
          data-testid="use-spring-carousel-item-wrapper"
          style="display: flex; flex: 1 0 calc(100% / 1); position: relative;"
        >
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            Item 4
          </div>
        </div>
        <div
          data-testid="use-spring-carousel-item-wrapper"
          style="display: flex; flex: 1 0 calc(100% / 1); position: relative;"
        >
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            Item 5
          </div>
        </div>
        <div
          data-testid="use-spring-carousel-item-wrapper"
          style="display: flex; flex: 1 0 calc(100% / 1); position: relative;"
        >
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            Item 6
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
